contrib/initramfs/scripts/Makefile.am

```diff
@@ -0,0 +1,19 @@
+scriptsdir = $(datarootdir)/initramfs-tools/scripts
+
+scripts_SCRIPTS = \
+	zfs
+
+EXTRA_DIST = \
+	$(top_srcdir)/contrib/initramfs/scripts/zfs.in
+	$(top_srcdir)/contrib/initramfs/scripts/local-top/zfs
```

Comment:
What's with the `local-top` entry?  This shouldn't be needed.

Reply:
I actually don't know how to properly use autotools, i just copy-pasted most of these files from the other dirs.

If i understand correctly `local-top/zfs` is needed in either `contrib/initramfs/scripts/Makefile.am` or `contrib/initramfs/Makefile.am`, not both; since `contrib/initramfs/scripts/Makefile.am` is dealing with files in "contrib/initramfs/scripts" maybe it's better to remove `local-top/zfs` from the other Makefile.am?

Reply:
The `EXTRA_DIST` variable is used to add additional files to the tarball which otherwise would be excluded because they're not directly referenced in a `Makefile.am` as `_SOURCES`, `_HEADERS`, `_SCRIPTS`, etc.

I think we have two pretty reasonable options here.

1) Keep everything needed to build and install this directory in one `contrib/initramfsMakefile.am` as it is now.  You'd then need to add the new `.in` targets to `contrib/initramfs/Makefile.am`.  Or,

2) Create a `Makefile.am` in each sub-directory which is responsible for only including, building, and installing the files in that directory.  This would be my preference and basically the way you were going with this.

Reply:
Does option 2 also include sub-directories? In this case it's better to leave `local-top/zfs` in this Makefile ("contrib/initramfs/scripts/Makefile.am"), right?

Reply:
Yes, you'd either want a `Makefile.am` in each sub-directory which is responsible for the files in that directory.  Like most of the rest of the tree.

```
initramfs/Makefile.am
initramfs/conf-hooks.d/Makefile.am
initramfs/hooks/Makefile.am
initramfs/scripts/Makefile.am
initramfs/scripts/local-top/Makefile.am
```

or, a single `initramfs/Makefile.am` responsible for all the sub-directories as it is now.

Either are reasonable options but we should avoid doing something in between which would be confusing.